Source/Xamarin/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -124,10 +124,18 @@ public virtual Task NavigateAsync(Uri uri, NavigationParameters parameters = nul
             }
             else if (currentPage is TabbedPage)
             {
+                var options = currentPage as IMultiPageNavigationOptions;
```

Comment:
This can't go here. It must be added when the page is created, not when it is being navigated away from